### PR TITLE
feat(CL): swap router module

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #3049

## What is the purpose of the change

Implementing new barebones swaprouter module

Context: https://github.com/osmosis-labs/osmosis/issues/3049

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: x/swaprouter/types/params.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+	_ paramtypes.ParamSet = &Params{}
+)
+
+const (
+	defaultPruneEpochIdentifier    = "day"
+	defaultRecordHistoryKeepPeriod = 48 * time.Hour
+)
+
+// ParamTable for twap module.
```
Comment (p0mvn) [inline x/swaprouter/types/params.go:9]:
```suggestion
// ParamTable for swaprouter module.
```

File: x/swaprouter/types/params.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+func ParamKeyTable() paramtypes.KeyTable {
+	return paramtypes.NewKeyTable().RegisterParamSet(&Params{})
+}
+
+func NewParams(pruneEpochIdentifier string, recordHistoryKeepPeriod time.Duration) Params {
+	return Params{}
+}
+
+// default twap module parameters.
```
Comment (p0mvn) [inline x/swaprouter/types/params.go:18]:
```suggestion
// DefaultParams returns default swaprouter module parameters.
```

File: proto/osmosis/swaprouter/v1beta1/genesis.proto [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+option go_package = "github.com/osmosis-labs/osmosis/v12/x/swaprouter/types";
+
+// Params holds parameters for the swaprouter module
+message Params {}
+
+// GenesisState defines the swaprouter module's genesis state.
+message GenesisState {
+  // params is the container of swaprouter parameters.
+  Params params = 2 [ (gogoproto.nullable) = false ];
```
Comment (mattverse) [inline proto/osmosis/swaprouter/v1beta1/genesis.proto:17]:
```suggestion
  Params params = 1 [ (gogoproto.nullable) = false ];
```

**Review by czarcas7ic — APPROVED** [review]

**Review by mattverse — APPROVED** [review]
LGTM once proto field is changed

**Pull request merged**